pywbem/_logging.py

```diff
@@ -14,371 +14,303 @@
+    handler.setFormatter(formatter)
+    logger.addHandler(handler)
 
-            ``log_spec`` := ``log_comp`` ['=' [ ``dest`` ][":"[ ``detail_level`` ]]]]
+    WBEMConnection.configure_api_logger(detail_level='summary', connection=True)
 
-            where:
+    # all of these connections do api call and response logging with
+    # summary information.
+    conn.WBEMConnection(...)
+    conn2.WBEMConnection(...)
+
+
+Configure the api logger for summary information output using only pywbem
+methods to configure the logger::
 
-            * ``log_comp``: Must be one of strings in the
-              :data:`~pywbem._logging.LOG_COMPONENTS` list.
+    WBEMConnection.configure_api_logger(dest='stderr',
+                                     detail_level='summary', connection=True)
 
-            * ``detail_level``: Must be one of strings in the
-              :data:`~pywbem._logging.LOG_DETAIL_LEVELS` list.
+    conn = WBEMConnection()
 
-            * ``dest``: Must be one of strings in the
-              :data:`~pywbem._logging.LOG_DESTINATIONS` list.
+Configure just a single WBEMConnection for output of summary information::
 
-          log_filename (:term:`string`)
-            Optional string that defines the filename for output of logs
-            if the dest type is `file`
+    conn = WBEMConnection()
+    conn.configure_api_logger(dest='file', log_filname='xxx.log',
+                              connection=conn)
 
-        Raises:
+Log everything to stderr and output all of the information using the
+pywbem method that configures the loggers from a string::
 
-          ValueError: Generated if the syntax of the input string is invalid
-            or any of the components is not one of allowed strings.
+    define_loggers_from_string('all=stderr':summary', connection=True)
+    conn = WBEMConnection()
 
-        Examples::
+or::
 
-            ops=stderr:min    # set cim operations logger
+    define_loggers_from_string('api=stderr':summary,http=summary',
+                               connection=True)
+    conn = WBEMConnection()
 
-            http=file:        # set http logger to send to file
+Log only http requests and responses including all infomration in the
+requests and responses to a file::
 
-            all=file:all      # Set all loggers to default log destination
-        """  # noqa: E501
-        # pylint: enable=line-too-long
+    define_logger('http', 'file', log_filename='mylog.log')
+    conn = WBEMConnection(..., enable_log='all')
 
-        results = cls._parse_log_specs(input_str)
-        for name, value in six.iteritems(results):
-            cls.create_logger(name, log_dest=value[0],
-                              log_detail_level=value[1],
-                              log_filename=log_filename)
+The logger names themselves are uniquely identified with an id string for
+each WBEMconnection so that the log itself may look like  the followingfor log
+configuration with detail_level = 'summary'::
```

Comment:
That should be described more explicitly. It is not clear what the id string is in the examples below. The format string for the example should be stated.
Also, we should introduce and use a consistent term for that id. How about "connection id" ?

Reply:
Modified this description to clarify that the logger names are pywbem.xxx.conn_id